plugins/patchmanager.koplugin/main.lua

```diff
@@ -0,0 +1,191 @@
+                    break
+                end
+            end
+        end
+    end
+
+    for priority = tonumber(userPatch.early_once), tonumber(userPatch.on_exit) do
+        table.sort(self.patches[priority], userPatch.sorting)
+    end
+end
+
+function PatchManager:getSubMenu(priority)
+    if #self.patches == 0 then
+        return {}
+    end
+    local function getExecutionStatus(patch_name)
+        return userPatch.execution_status[patch_name] == false and " ⚠" or ""
+    end
+    local sub_menu = {}
+    for i, patch in ipairs(self.patches[priority]) do
+        local ext = ".lua"
+        -- strip anything after ".lua" in patch_name
+        local patch_name = patch
+        patch_name = patch_name:sub(1, patch_name:find(ext, 1, true) + ext:len() - 1)
+        table.insert(sub_menu, {
+            text = patch_name .. getExecutionStatus(patch_name),
+            checked_func = function()
+                return patch:find("%.lua$") ~= nil
+            end,
+            callback = function()
+                local extension_pos = patch:find(ext, 1, true)
+                if extension_pos then
+                    local is_patch_enabled = extension_pos == patch:len() - (ext:len() - 1)
+                    if is_patch_enabled then -- patch name ends with ".lua"
+                        local disabled_name = patch .. self.disable_ext
+                        os.rename(self.patch_dir .. "/" .. patch,
+                                  self.patch_dir .. "/" .. disabled_name)
+                        patch = disabled_name
+                    else -- patch name name contains ".lua"
+                        local enabled_name = patch:sub(1, extension_pos + ext:len() - 1)
+                        os.rename(self.patch_dir .. "/" .. patch,
+                                  self.patch_dir .. "/" .. enabled_name)
+                        patch = enabled_name
+                    end
+                end
+                UIManager:askForRestart(T(
+                    _("Patches changed. %1\n"),
+                    _("Current set of patches will be applied on next restart.")))
+            end,
+            hold_callback = function()
+                local patch_fullpath = self.patch_dir .. "/" .. patch
+                if self.ui.texteditor then
+                    local function done_callback()
+                        UIManager:askForRestart(T(
+                            _("Patches might have changed. %1\n"),
+                            _("Current set of patches will be applied on next restart.")))
+                    end
+                    self.ui.texteditor:quickEditFile(patch_fullpath, done_callback, false)
+                else -- fallback to show only the first lines
+                    local patch_lines = {}
+                    for line in io.lines(patch_fullpath) do
+                        table.insert(patch_lines, line)
+                    end
+                    local patch_content = table.concat(patch_lines, "\n")
```

Comment:
```lua
    local file = io.open(file_path, "rb")
    if not file then
        return ""
    end
    local file_content = file:read("*all")
    file:close()
```

See TextEditor:readFileContent(file_path) for the normal way of reading a file content :)
(unless this is on purpose and Lua io.lines somehow accepts and clean various linux/windows/mac line endings ? - dunno if I checked that when I made TextEditor :/)